packages/expression-evaluator/lib/types.ts

```diff
@@ -0,0 +1,23 @@
+import type * as RDF from '@rdfjs/types';
+import type * as E from './expressions';
+import type { IInternalEvaluator } from './functions/OverloadTree';
+
+/**
+ * An evaluator for RDF expressions.
+ */
+export interface IExpressionEvaluator extends IInternalEvaluator {
+  /**
+   * Evaluates the provided bindings in terms of the context the evaluator was created.
+   * @param mapping the RDF bindings to evaluate against.
+   */
+  evaluate: (mapping: RDF.Bindings) => Promise<RDF.Term>;
+
+  /**
+   * Evaluates the provided bindings in terms of the context the evaluator was created,
+   * returning the effective boolean value.
+   * @param mapping the RDF bindings to evaluate against.
+   */
+  evaluateAsEBV: (mapping: RDF.Bindings) => Promise<boolean>;
+
+  evaluateAsInternal: (mapping: RDF.Bindings) => Promise<E.Expression>;
```

Comment:
If it's only supposed to be used internally, is it needed on this interface?

Reply:
Apparently not anymore :)

Reply:
Okey, This PR has been open for too long. I think the internal is a wrong name. I's more like the thing this returns has been parsed by the expression evaluator. So maybe `evaluateAsEvaluatorExpression` would be better?

Reply:
That works for me!